### PR TITLE
Account for RecordKeeper's hash_map in def record_group_name of logging_presets.py

### PR DESCRIPTION
**Pull request by HSinger04**
Fix issue described in two first comments of #591.

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
Thanks @HSinger04

File: src/pytorch_metric_learning/utils/logging_presets.py [hunk 1]
```diff
@@ -365,7 +365,12 @@ def record_group_name(self, tester, split_name):
         )
         if reference_sets_label == query_set_label:
             reference_sets_label = "self"
-        return f"{base_record_group_name}_{query_set_label}_vs_{reference_sets_label}"
+        record_group_name = f"{base_record_group_name}_{query_set_label}_vs_{reference_sets_label}"
+        if record_group_name in self.record_keeper.hash_map.values():
```
Comment (KevinMusgrave) [inline src/pytorch_metric_learning/utils/logging_presets.py:369]: I think we can get rid of this if statement

File: src/pytorch_metric_learning/utils/logging_presets.py [hunk 2]
```diff
@@ -365,7 +365,12 @@ def record_group_name(self, tester, split_name):
...
+            for key, val in self.record_keeper.hash_map.items():
+                if val == record_group_name:
+                    record_group_name = key
```
Comment (KevinMusgrave) [inline src/pytorch_metric_learning/utils/logging_presets.py:372]: Add `break` after a match is found

File: src/pytorch_metric_learning/utils/logging_presets.py (hunk 1 above)
Comment (HSinger04) [reply src/pytorch_metric_learning/utils/logging_presets.py:369]: Yes, true

**Comment by HSinger04** [comment]
@KevinMusgrave Incorporated the changes you suggested.

**Pull request merged**